openmdao/core/problem.py

```diff
@@ -1059,15 +1108,43 @@ def _calc_gradient_ln_solver(self, indep_list, unknown_list, return_format, mode
                                     if J[item][param] is None:
                                         J[item][param] = np.zeros((nk, len(in_idxs)))
                                     J[item][param][:, j-jbase] = dxval
+
+                                    # Driver scaling
+                                    if param in in_scale:
+                                        J[item][param][:, j-jbase] *= in_scale[param]
+                                    if item in un_scale:
+                                        J[item][param][:, j-jbase] *= un_scale[item]
```

Comment:
the in_scale and un_scale cases are red for coverage... would it be possible to get a serial test case for these